src/cli.ts

```diff
@@ -0,0 +1,183 @@
+};
+
+const setAzureAccessToken = async () => {
+    const promptData: Answers<PromptType> = await prompts({
+        name: PROMPT_NAME,
+        type: "text",
+        message: "Enter your Azure personal access token: ",
+        hint: "e.g. 3Ccz4G6QPilk",
+        validate: (val) => {
+            if (val.length === 0) {
+                return "You must enter a value";
+            } else {
+                return true;
+            }
+        },
+    });
+
+    return promptData[PROMPT_NAME];
+};
+
+const setGitlabBaseURL = async () => {
+    const promptData: Answers<PromptType> = await prompts({
+        name: PROMPT_NAME,
+        type: "text",
+        message: "Enter your Gitlab organisation's base URL: ",
+        hint: "e.g. https://gitlab.example.com/",
+        validate: (val) => {
+            if (val.length === 0) {
+                return "You must enter a value";
+            } else {
+                return true;
+            }
+        },
+    });
+
+    return promptData[PROMPT_NAME];
+};
+
+const setGitlabAccessToken = async () => {
+    const promptData: Answers<PromptType> = await prompts({
+        name: PROMPT_NAME,
+        type: "text",
+        message: "Enter your Gitlab personal access token: ",
+        hint: "e.g. Hf4sXcfn7M69",
+        validate: (val) => {
+            if (val.length === 0) {
+                return "You must enter a value";
+            } else {
+                return true;
+            }
+        },
+    });
+
+    return promptData[PROMPT_NAME];
+};
+
+const getConfigFromCli = async (): Promise<void> => {
+    try {
+        const startDate: Date = await setStartDate();
+        const endDate: Date = await setEndDate(startDate);
+        const orgs: string[] = await setOrganisations();
+        const azureEnabled = orgs.includes("Azure");
+        const gitlabEnabled = orgs.includes("Gitlab");
+
+        overrideConfig({
+            startDate: startDate,
+            endDate: endDate,
+            azure: { ...getConfig().azure, enabled: azureEnabled },
+            gitlab: { ...getConfig().gitlab, enabled: gitlabEnabled },
+        });
+
+        if (azureEnabled) {
+            const azureBaseUrl: string = await setAzureBaseURL();
+            const azureAccessToken: string = await setAzureAccessToken();
+
+            overrideConfig({
+                azure: {
+                    ...getConfig().azure,
+                    baseURL: azureBaseUrl,
+                    personalAccessToken: azureAccessToken,
+                },
+            });
+        }
+
+        if (gitlabEnabled) {
+            const gitlabBaseUrl: string = await setGitlabBaseURL();
+            const gitlabAccessToken: string = await setGitlabAccessToken();
+
+            overrideConfig({
+                gitlab: {
+                    ...getConfig().gitlab,
+                    baseURL: gitlabBaseUrl,
+                    personalAccessToken: gitlabAccessToken,
+                },
+            });
+        }
```

Comment:
This should be mostly extracted to another method, which is then executed within this try/catch